File: src/lib/dexie/models/account.ts
import type { CashCompassDexie } from "..";

export type Account = {
	id: string;
	label: string;
	accountType: 'expenses' | 'net_worth';
	isArchived: 0 | 1;
};

export const AccountUtils = {
	getUnarchived: (db: CashCompassDexie) => db.account.filter(a => !a.isArchived),
	getArchived: (db: CashCompassDexie) => db.account.filter(a => !!a.isArchived)
};

File: server/internal/model/account.go
package model

import "time"

// AccountType represents the two valid account classifications.
type AccountType string

const (
	AccountTypeExpenses AccountType = "expenses"
	AccountTypeNetWorth AccountType = "net_worth"
)

// IsValid reports whether at is a recognised AccountType.
func (at AccountType) IsValid() bool {
	return at == AccountTypeExpenses || at == AccountTypeNetWorth
}

type Account struct {
	ID          int
	Label       string
	AccountType AccountType
	IsArchived  bool
	CreatedAt   time.Time
	UpdatedAt   time.Time
}
